factory/__init__.py

```diff
@@ -1,6 +1,8 @@
 # -*- coding: utf-8 -*-
 # Copyright: See the LICENSE file.
 
+import pkg_resources
```

Comment:
Just an FYI. Starting with Python 3.8, grabbing a package version will be supported by the stdlib through `importlib.metadata.version()`. Just thought I'd mention it in case you'd like to add a comment or fallback.

https://docs.python.org/3/library/importlib.metadata.html#distribution-versions